changelog.d/2592.change.rst

```diff
@@ -0,0 +1 @@
+Warned for usage of uppercase keys in :code:`metadata` in :code:`setup.cfg`, restored some compatiblity by transforming to lowercase if not. Followed the change of keys in :code:`setup.cfg` now being case-sensitive -- by :user:`melissa-kun-li`
```

Comment:
FYI RST has double backticks for inline code so there's no need to use an explicit `:code:` role. Also, according to the code, the user-facing change (the impact) is that the option name is normalized, not that the warning is emitted — that warning is additional instrumentation, not the main change:

```suggestion
Made option keys in the ``[metadata]`` section of ``setup.cfg`` case-sensitive. Users having
uppercase option spellings will get a warning suggesting to make them to lowercase
-- by :user:`melissa-kun-li`
```